server/mediaserver.go

```diff
@@ -578,6 +585,11 @@ func getRTMPStreamHandler(s *LivepeerServer) func(url *url.URL) (stream.RTMPVide
 
 // HandlePush processes request for HTTP ingest
 func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
+	if !s.HTTPIngest {
```

Comment:
Rather than checking here, should we just not register the path, and allow the framework to take care of the error handling? Eg,

https://github.com/livepeer/go-livepeer/blob/c6b06679fe4adaa204c6d0a9628e184b2fadc0b6/server/mediaserver.go#L132-L134

Changes to:

```
if lpNode.NodeType == core.BroadcasterNode && HTTPIngest {
 	opts.HttpMux.HandleFunc("/live/", ls.HandlePush) 
 } 
```

Up to you how to propagate in the `HTTPIngest` variable. Could be via a global in `mediaserver.go` or add an extra param to `NewLivepeerServer`. Slight preference for the latter, since that's less state (whether globally or in `LivepeerServer`).

Reply:
Fixed in 94ee818